AutoNode/validator.py

```diff
@@ -297,101 +339,162 @@ def activate_validator():
         all_val = staking.get_all_validator_addresses(endpoint=node_config['endpoint'])
         if validator_config["validator-addr"] in all_val:
             log(f"{Typgpy.OKBLUE}Activating validator{Typgpy.ENDC}")
-            response = cli.single_call(
-                ['hmy', 'staking', 'edit-validator', '--validator-addr', f'{validator_config["validator-addr"]}',
-                 '--active', 'true', '--node', f'{node_config["endpoint"]}',
-                 '--passphrase-file', saved_wallet_pass_path, '--gas-price', f'{validator_config["gas-price"]}'])
+            passphrase = get_wallet_passphrase()
+            cmd = ['hmy', 'staking', 'edit-validator', '--validator-addr', f'{validator_config["validator-addr"]}',
+                   '--active', 'true', '--node', f'{node_config["endpoint"]}',
+                   '--passphrase']
+            if validator_config["gas-price"]:
+                cmd.extend(['--gas-price', f'{validator_config["gas-price"]}'])
+            proc = cli.expect_call(cmd)
+            interact_wallet_passphrase(proc, passphrase)
+            proc.expect(pexpect.EOF)
+            response = proc.before.decode()
             log(f"{Typgpy.OKGREEN}Edit-validator response: {response}{Typgpy.ENDC}")
         else:
             log(f"{Typgpy.FAIL}Address {validator_config['validator-addr']} is not a validator!{Typgpy.ENDC}")
     except (TimeoutError, ConnectionError, RuntimeError, subprocess.CalledProcessError) as e:
         log(traceback.format_exc())
         log(f"{Typgpy.FAIL}{Typgpy.BOLD}Edit-validator error: {e}{Typgpy.ENDC}")
-        if not _recover_interaction:
+        if not _hard_reset_recovery:
             raise e
         log(f"{Typgpy.WARNING}{Typgpy.BOLD}Continuing...{Typgpy.ENDC}")
 
 
-def setup(recover_interaction=False):
+def collect_reward():
+    try:
+        all_val = staking.get_all_validator_addresses(endpoint=node_config['endpoint'])
+        if validator_config["validator-addr"] in all_val:
+            log(f"{Typgpy.OKBLUE}Collecting rewards{Typgpy.ENDC}")
+            passphrase = get_wallet_passphrase()
+            cmd = ['hmy', 'staking', 'collect-rewards', '--delegator-addr', validator_config['validator-addr'],
+                   '--node', f'{node_config["endpoint"]}', '--passphrase']
+            if validator_config["gas-price"]:
+                cmd.extend(['--gas-price', f'{validator_config["gas-price"]}'])
+            proc = cli.expect_call(cmd)
+            interact_wallet_passphrase(proc, passphrase)
+            proc.expect(pexpect.EOF)
+            response = proc.before.decode()
+            log(f"{Typgpy.OKGREEN}Collect rewards response: {response}{Typgpy.ENDC}")
+        else:
+            log(f"{Typgpy.FAIL}Address {validator_config['validator-addr']} is not a validator!{Typgpy.ENDC}")
```

Comment:
Not a delegator

Reply:
collect rewards ONLY applies to validators, in the context of autonode.